### PR TITLE
CONSOLE-2431: i18n for home/api-explorer

### PR DESCRIPTION
**Pull request by jcaianirh**
For: https://issues.redhat.com/browse/CONSOLE-2431

Addresses api explorer section in the home navigation

### CODE REVIEW

File: frontend/public/components/api-explorer.tsx [hunk 1]
```diff
@@ -339,48 +348,56 @@ const APIResourcesList = compose(
...
-    </div>
-    <APIResourcesList />
-  </>
-);
+export const APIExplorerPage: React.FC<{}> = () => {
+  const { t } = useTranslation();
+  return (
+    <>
+      <Helmet>
+        <title>{t('api-explorer~Explore API Resources')}</title>
```
Comment (rebeccaalpert) [inline frontend/public/components/api-explorer.tsx:356]: Should "Resources" be lowercase here the way it is above?

File: frontend/public/components/api-explorer.tsx [hunk 2]
```diff
@@ -339,48 +348,56 @@ const APIResourcesList = compose(
...
+      </Helmet>
+      <div className="co-m-nav-title">
+        <h1 className="co-m-pane__heading">{t('api-explorer~Explore API Resources')}</h1>
```
Comment (rebeccaalpert) [inline frontend/public/components/api-explorer.tsx:359]: Should "Resources" be lowercase here the way it is above?

File: frontend/public/components/api-explorer.tsx [hunk 3]
```diff
@@ -578,35 +601,37 @@ const APIResourceAccessReview: React.FC<APIResourceTabProps> = ({
...
           <CheckBox
-            title="Group"
+            title={t('api-explorer~Group')}
             active={showGroups}
             number={groups.length}
             toggle={toggleShowGroups}
           />
           <CheckBox
-            title="ServiceAccount"
+            title={t('api-explorer~ServiceAccount')}
```
Comment (rebeccaalpert) [inline frontend/public/components/api-explorer.tsx:616]: It's not clear to me whether we should i18n these. I'm sure Sam will chime in here.

**Review by rebeccaalpert — CHANGES_REQUESTED** [review]
Looks good to me! Just a couple of capitalization questions.

File: frontend/public/components/api-explorer.tsx (hunk 1 above)
Comment (jcaianirh) [reply frontend/public/components/api-explorer.tsx:356]: @rebeccaalpert this is a page title, so i think its capitalized.

File: frontend/public/components/api-explorer.tsx (hunk 2 above)
Comment (jcaianirh) [reply frontend/public/components/api-explorer.tsx:359]: @rebeccaalpert this is a page title, so i think its capitalized.

File: frontend/public/components/api-explorer.tsx (hunk 1 above)
Comment (rebeccaalpert) [reply frontend/public/components/api-explorer.tsx:356]: Perfect! Looks good to me then.

File: frontend/public/components/api-explorer.tsx (hunk 2 above)
Comment (rebeccaalpert) [reply frontend/public/components/api-explorer.tsx:359]: Perfect! Lgtm then.

File: frontend/public/components/api-explorer.tsx (hunk 3 above)
Comment (jcaianirh) [reply frontend/public/components/api-explorer.tsx:616]: @rebeccaalpert I think last i heard these are left as is...

**Review by rebeccaalpert — APPROVED** [review]

File: frontend/public/components/api-explorer.tsx [hunk 4]
```diff
@@ -578,35 +601,37 @@ const APIResourceAccessReview: React.FC<APIResourceTabProps> = ({
...
-          The following subjects can {verb} {plural}
-          {namespaced && namespace && <> in namespace {namespace}</>}
-          {namespaced && !namespace && <> in all namespaces</>}
-          {!namespaced && <> at the cluster scope</>}.
+          {t('api-explorer~The following subjects can {{verb}} {{plural}}', { verb, plural })}
+          {namespaced && namespace && (
+            <> {t('api-explorer~in namespace {{ namespace }}', { namespace })}</>
+          )}
+          {namespaced && !namespace && <> {t('api-explorer~in all namespaces')}</>}
+          {!namespaced && <> {t('api-explorer~at the cluster scope')}</>}.
```
Comment (spadgett) [inline frontend/public/components/api-explorer.tsx:628]: The entire sentence needs to be one message with variables. (I realize we might need a couple different messages here depending on whether it's namespaced.)

Comment (jcaianirh) [reply frontend/public/components/api-explorer.tsx:628]: ok, @spadgett created 3 full messages based on logical differences.

**Review by spadgett — APPROVED** [review]
/approve
LGTM, please squash

/hold for other approvals
/assign @yapei @ahardin-rh @sferich888

**Comment by yapei** [comment]
@jcaianirh I created a cluster with command `launch openshift/console#7009`  then I visited api-explorer page with two methods:
1. Adding `?lng=ja` suffix to URL which is `<console_route>/api-explorer?lng=ja` 
2. Changed browser setting to `ja` and refresh the <console_route>/api-explorer page 

However neither way shows me the Japanese translation, nothing is translated

**Comment by jcaianirh** [comment]
@yapei These strings have not yet been translated into any language. You can see the changes by using pseudolocalization url param

**Comment by yapei** [comment]
> @yapei These strings have not yet been translated into any language. You can see the changes by using pseudolocalization url param

`using pseudolocalization url param` does it mean adding `?lng=en` suffix or something?

**Comment by rebeccaalpert** [comment]
Hey @yapei - `?pseudolocalization=true&lng=en` is the correct parameter.

**Comment by yapei** [comment]
@jcaianirh  Is the string `Select all filters` should be translated?
Screen Shot 2020-11-23 at 11 33 57 AM

**Comment by jcaianirh** [comment]
@yapei thanks, this is a common component, the common components will be addressed as a separate pr. The first 3 items are passed in by the api-explorer, and the "Select all filters" is provided by the common component. It will be translated when the common components are translated.

**Comment by rebeccaalpert** [comment]
We've already completed/merged the common components story and this was not included. I added a comment to https://issues.redhat.com/browse/CONSOLE-2474 about this piece of text so it doesn't get lost.

**Comment by jcaianirh** [comment]
@yapei this is ready for qe review, the note you added about select all filters will be fixed in another pr. thanks!

**Comment by yapei** [comment]
Thank you all @jcaianirh  @rebeccaalpert 
I understand that the issue I mentioned will come in another PR

**Review by spadgett — APPROVED** [review]
/lgtm

Leaving the hold, however, since we're over our SLO. We can remove the hold when our bug count drops.

**Comment by spadgett** [comment]
Removing the hold since we're below our SLO.

/hold cancel

**Pull request merged**